### PR TITLE
RSDK-7565, RSDK-7471: add new methods get_properties and capture_all_from_camera

### PR DESCRIPTION
**Pull request by bhaney**
Adds two new methods, and tests for them

- GetProperties
- CaptureAllFromCamera

Adding it to the rc-0.21.0 branch

### CODE REVIEW

File: src/viam/services/vision/vision.py [hunk 1]
```diff
@@ -1,27 +1,106 @@
...
+
+        ::
+
+            camera_name = "cam1"
+
+            # Grab the detector you configured on your machine
+            my_detector = VisionClient.from_robot(robot, "my_detector")
+
+            # capture all from the next image from the camera
+            request = vision.CaptureAllRequest(return_image=True, return_detections=True)
```
Comment (njooma) [inline src/viam/services/vision/vision.py:90]: This is no longer accurate

File: src/viam/services/vision/vision.py [hunk 2]
```diff
@@ -1,27 +1,106 @@
...
+            result = await my_detector.capture_all_from_camera(camera_name, request)
+
+        Args:
+            camera_name (str): The name of the camera to use for detection
+            request (vision.CaptureAllRequest): What the vision service should return from CaptureAllFromCamera
```
Comment (njooma) [inline src/viam/services/vision/vision.py:95]: Update args

**Review by njooma — APPROVED** [review]

File: src/viam/services/vision/vision.py [hunk 3]
```diff
@@ -1,27 +1,106 @@
...
     )
 
+    Properties: "TypeAlias" = GetPropertiesResponse
+    """
+    Properties is a class that states what features are supported on the associated vision service.
+    Currently, these are the following properties:
+    classifications_supported (bool): GetClassifications and GetClassificationsFromCamera are implemented.
+    detections_supported (bool): GetDetections and GetDetectionsFromCamera are implemented.
+    object_point_clouds_supported (bool): GetObjectPointClouds is implemented.
+    """
```
Comment (bhaney) [inline src/viam/services/vision/vision.py:65]: Will this show up in the python docs online?

**Review by njooma — APPROVED** [review]
Just a few changes to documentation, otherwise this looks good! No need to re-request review once the docs are fixed

File: src/viam/services/vision/vision.py (hunk 3 above)
Comment (njooma) [reply src/viam/services/vision/vision.py:65]: This hsould be fine. You might want to indent/add a `*` or `-` to the attributes

Comment (bhaney) [reply src/viam/services/vision/vision.py]: linter actually complains about an unexpected indent, but I did add the `-`

**Pull request merged**